server/controllers/auth_controller.coffee

```diff
@@ -17,6 +18,46 @@ auth =
       # return same message regardless of login problem
       return done null, false, { message: 'Incorrect login.' }
 
+  google: (req, res) ->
+    auth_code = req.body.code
+    client_id = req.body.clientId
+    redirectUri = req.body.redirectUri
+    secret = config.googleAuthConfig.clientSecret
+
+    # Buil Oauth for Google
+    OAuth2 = googleapis.auth.OAuth2
+    oauth2Client = new OAuth2 client_id, secret, redirectUri
+    oauth2Client.getToken auth_code, (err, tokens)->
+      if err
+        res.status(500).send err
+      else
+        # Make API request to google for user info
+        oauth2Client.setCredentials(tokens)
+        oauth2 = googleapis.oauth2('v2')
+        oauth2.userinfo.get {
+        auth: oauth2Client
+        }, (err, google_user) ->
+          # Save user & auth information in user object
+          google_user.token = createToken(google_user)
+          google_user.auth = tokens
+          User.methods.findOne { id: google_user.id}, (err, user) ->
+            if err
+              res.status(500).send err
+            else
+              if user
+                user.token = google_user.token
+                response = buildAuthResponse(user)
+                res.status(200).send response
+              else
+                User.methods.addUser google_user, (err, result) ->
```

Comment:
everything we want to save, at least -- thats google user info, token, and goolgleapis auth info